### PR TITLE
Upgrade NuGet package versions to resolve Native AOT warnings.

### PR DESCRIPTION
**Pull request by IEvangelist**
- Upgrade NuGet package versions to resolve Native AOT warnings.
- Update copyright dates.
- Remove `strategy: fail-fast: false` as the workflow isn't a matrix.
- Update to `actions/checkout@v3`.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -152,7 +152,7 @@ csharp_space_between_parentheses = false
 csharp_space_between_square_brackets = false
 
 # License header
-file_header_template = Copyright (c) GitHub 2023 � Licensed as MIT.
+file_header_template = Copyright (c) GitHub 2023-2024 � Licensed as MIT.
```
Comment (nickfloyd) [inline .editorconfig:155]: Thanks for taking care of the 🏕️

File: src/GitHub.Octokit.SDK.csproj [hunk 2]
```diff
@@ -4,6 +4,7 @@
     <TargetFramework>net8.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
+    <PublishAot>true</PublishAot>
```
Comment (nickfloyd) [inline src/GitHub.Octokit.SDK.csproj:7]: Thank you for this as well. I missed it in the original drop. ❤️

**Review by nickfloyd — APPROVED** [review]
Hey @IEvangelist, as always, thank you for the efforts here ❤️

**Review by kfcampbell — APPROVED** [review]
:rocket: :rocket: :rocket:

**Comment by nickfloyd** [comment]
@IEvangelist Those merge conflicts are on me.  I'd be glad to clean them up if you cannot get to them.  Apologies for the disruption there.

**Comment by IEvangelist** [comment]
> @IEvangelist Those merge conflicts are on me. I'd be glad to clean them up if you cannot get to them. Apologies for the disruption there.

@nickfloyd - No worries, I resolved them - thank you

**Pull request merged**